library/src/main/java/com/feedhenry/sdk/Sync.java

```diff
@@ -0,0 +1,145 @@
+import com.feedhenry.sdk.android.NetworkClientImpl;
+import com.feedhenry.sdk.android.SyncableActivity;
+import com.feedhenry.sdk.network.NetworkClient;
+import com.feedhenry.sdk.sync.FHSyncClient;
+import com.feedhenry.sdk.sync.FHSyncConfig;
+import okhttp3.MediaType;
+
+/**
+ * The Sync class provides static methods to initialize the library, create new instances of all the
+ * API request objects, and configure global settings.
+ */
+public class Sync {
+
+    private static final String TAG = "FHSync";
+
+    public static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+    public static final int LOG_LEVEL_VERBOSE = 1;
+    public static final int LOG_LEVEL_DEBUG = 2;
+    public static final int LOG_LEVEL_INFO = 3;
+    public static final int LOG_LEVEL_WARNING = 4;
+    public static final int LOG_LEVEL_ERROR = 5;
+    public static final int LOG_LEVEL_NONE = Integer.MAX_VALUE;
+
+    private static int mLogLevel = LOG_LEVEL_ERROR;
+
+    private static NetworkClient networkClient;
+
+    private Sync() {
+    }
+
+    /**
+     * Sync initialization.
+     * @param application application object
+     * @param config sync config
+     * @param cloudUrl url of the cloud app endpoint
+     */
+    public static void init(final Application application, final FHSyncConfig config, final String cloudUrl) {
+        final Context appContext = application.getApplicationContext();
+        networkClient = new NetworkClientImpl(appContext,cloudUrl,new ClientIdGeneraorImpl(appContext));
+        networkClient.registerNetworkListener();
+        application.registerActivityLifecycleCallbacks(new Application.ActivityLifecycleCallbacks() {
+            @Override
+            public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+                if (activity instanceof SyncableActivity) {
+                    Log.d(TAG, "sync init for " + activity.getClass().getName());
+                    FHSyncClient.getInstance().init(config, new FileStorageImpl(appContext),networkClient);
+                    FHSyncClient.getInstance().setListener(((SyncableActivity) activity).onBindSyncListener());
+
+                }
+            }
+
+            @Override
+            public void onActivityStarted(Activity activity) {
+
+            }
+
+            @Override
+            public void onActivityResumed(Activity activity) {
+                if (activity instanceof SyncableActivity) {
+                    Log.d(TAG, "sync resumed for " + activity.getClass().getName());
+                    FHSyncClient.getInstance().resumeSync(((SyncableActivity) activity).onBindSyncListener());
+                }
+            }
+
+            @Override
+            public void onActivityPaused(Activity activity) {
+                if (activity instanceof SyncableActivity) {
+                    Log.d(TAG, "sync paused for " + activity.getClass().getName());
+                    FHSyncClient.getInstance().pauseSync();
```

Comment:
In this scenario, if there are 2 `SyncableActivity` instances (for example, each for a different dataset), will pause one of them cause the sync process stopped in the other activity?

Reply:
I will have to test this myself.  Generally in Android an application only has one activity running at a time, but with N and split screen mode it might be possible to have multiple.

However the design of the old SDK encouraged people to embed inside of a Activity a FHSyncListener as an anonymous inner class.  Generally AICs inside of an activity are wrong because they cause stability problems if they aren't handled properly.  What we are trying to do is automate as much of the problematic behavior away while leaving the ease of use.

Since we are breaking the library out I am in favor of being more aggressive with pruning the architecture.

Reply:
I agree. The old Sync implementation was implemented in the early days when I don't know too much about developing Android apps properly 😞 . Now looking back, I can certainly see the problems with the current implementation. 

I think now is a good time to refactor the code given that we want to release sync as a standalone module. I don't think we need to worry too much about backward compatibility during the refactoring, as long as we can make sure the standalone lib can be running together with the existing Android SDK - this will allow developers to upgrade their app if they prefer to do so.

Reply:
I think we still want backward compatibility at least little bit, if we plan to put it wrapped in fh-android-sdk. So the manual way of doing things directly with FHSyncClient should be still there, but for new developers we should do something better.